pkg/scheduler/apis/config/validation/validation_pluginargs.go

```diff
@@ -294,13 +296,16 @@ func ValidateNodeAffinityArgs(path *field.Path, args *config.NodeAffinityArgs) e
 
 // ValidateVolumeBindingArgs validates that VolumeBindingArgs are set correctly.
 func ValidateVolumeBindingArgs(path *field.Path, args *config.VolumeBindingArgs) error {
-	var err error
+	var allErrs field.ErrorList
 
 	if args.BindTimeoutSeconds < 0 {
-		err = field.Invalid(path.Child("bindTimeoutSeconds"), args.BindTimeoutSeconds, "invalid BindTimeoutSeconds, should not be a negative value")
+		allErrs = append(allErrs, field.Invalid(path.Child("bindTimeoutSeconds"), args.BindTimeoutSeconds, "invalid BindTimeoutSeconds, should not be a negative value"))
 	}
 
-	return err
+	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeCapacityPriority) {
+		allErrs = append(allErrs, validateFunctionShape(args.Shape, path.Child("shape"))...)
```

Comment:
@liggitt we don't have "field dropping" for config APIs right? In lieu of that, I think validating the field in all cases and ignoring it at runtime is the best approach.

Another approach of returning an error if the feature is disabled will break rollback/disable scenarios.

Reply:
> In lieu of that, I think validating the field in all cases and ignoring it at runtime is the best approach.

+1 and implemeneted

Reply:
I'd go a step further and forbid values in this config field if the feature is off... otherwise someone can specify a shape that is syntactically valid but does not behave at all like they want, successfully run the scheduler with this feature off, and then break themselves with the same config when the feature turns on in the future.

Reply:
If we return an error, then if someone has the feature on and sets the config, but then rolls back/disables the feature, then they would have to update their config at that time too otherwise the config will fail validation. Is that the expected approach for config apis?

Reply:
hmm... I don't love either direction, but in the rollback case, since they're already modifying the invocation, I think making them comment out the config for the disabled feature is better than letting people put timebomb inert configuration in the file and then activating it on upgrade when the feature becomes active

Reply:
it seems a safer choice, done in https://github.com/kubernetes/kubernetes/pull/99594/commits/3e788a3a7d4dc5de0c6e91e68bf755af8b2c4b45